src/python/pants/engine/native.py

```diff
@@ -545,8 +545,13 @@ def drop_handles(self, handles):
 
   def identify(self, obj):
     """Return an Ident-shaped tuple for the given object."""
-    hash_ = hash(obj)
-    type_id = self.to_id(type(obj))
+    try:
+      hash_ = hash(obj)
+      type_id = self.to_id(type(obj))
```

Comment:
Should this also be in the try, or outside? The comment talks just about hash

Reply:
We don't use this idiom much, but I recommend using the `else` clause for exactly this reason. The idea is `try` only has what might fail, `except` has error handling, and `else` has the rest of the related logic.

```python
try:
  hash_ = hash(obj)
except TypeError as e:
  raise TypeError("For object with type '{}', error was: {}".format(type(obj).__name__, e))
else:
  type_id = self.to_id(type(obj))
  return (hash_, TypeId(type_id))
```

Stack overflow post explaining why `try-except-else` is good practice and Pythonic: https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python.

Reply:
I was actually able to remove this change entirely by catching `TypeError` in the datatype `__hash__` method!